### PR TITLE
add a `--configuration` option to `pod lint`.

### PR DESCRIPTION
**Pull request by gereons**
This implements #9686

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* add a `--configuration` option to `pod lint`.  
  [gereons](https://github.com/gereons)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by gereons** [comment]
Tbh I have no clue what's causing the test failure on CircleCI. Are there any actionable errors in the log besides "CompileSwift failed with a nonzero exit code"?

**Comment by paulb777** [comment]
The CircleCI issue might be related to #9761?

Is it possible to add a test to spec/functional/command/repo/push_spec.rb?

**Comment by gereons** [comment]
If I understand #9761 correctly, it just removes the creation of a file that is ignored anyway, so I don't think it's related.

Also, I'm ot sure if .../repo/push_spec.rb is the right place for testing this? Looking at those files, ../lib/list_spec.rc seems to be more suitable to me. I can certainly add a test there.

File: lib/cocoapods/command/lib/lint.rb [hunk 1]
```diff
@@ -36,6 +36,7 @@ def self.options
             ['--skip-tests', 'Lint skips building and running tests during validation'],
             ['--test-specs=test-spec1,test-spec2,etc', 'List of test specs to run'],
             ['--analyze', 'Validate with the Xcode Static Analysis tool'],
+            ['--configuration=CONFIGURATION', 'Build using the given configuration (defaults to Release)'],
```
Comment (amorde) [inline lib/cocoapods/command/lib/lint.rb:39]: can we also add this to `spec lint`? unfortunately the options are duplicated at the moment and have to be added to both

**Comment by amorde** [comment]
Thanks for the contribution!

**Pull request merged**